Lexplorer/Pages/NFTCollection.razor

```diff
@@ -0,0 +1,174 @@
+                    Selected="@goToPage"
+                    Count="@goToNextPage"
+                    SelectedChanged="@((int page) => goToPage = page)"/>
+                 </MudToolBar>
+           </MudPaper>
+       </MudHidden>
+       <MudDivider DividerType="DividerType.Middle" Class="my-6"/>
+    </MudGrid>
+</MudContainer>
+
+@code {
+    [Parameter]
+    public string? tokenAddress { get; set; }
+
+    [Parameter]
+    [SupplyParameterFromQuery]
+    public string pageNumber { get; set; } = "1";
+
+    public int goToPage
+    {
+        get
+        {
+            return int.TryParse(pageNumber, out int np) ? np : 1;
+        }
+        set
+        {
+            pageNumber = value.ToString(); //set immediately for MudPagination to not switch forth and back
+            string URL = $"/collections/{tokenAddress}?pageNumber={value}";
+            NavigationManager.NavigateTo(URL);
+        }
+    }
+
+    public int goToNextPage
+    {
+        get
+        {
+            return int.TryParse(pageNumber, out int np) ? np + 1 : 1;
+        }
+    }
+
+    public readonly int pageSize = 12; //6 per row
+    private CancellationTokenSource? cts;
+    private IList<NonFungibleToken>? collectionNFTSlots { get; set; } = new List<NonFungibleToken>();
+    private Dictionary<string, NftMetadata> NFTdata { get; set; } = new Dictionary<string, NftMetadata>();
+    protected override async Task OnParametersSetAsync()
+    {
+        //cancel any previous OnParametersSetAsync which might still be running
+        cts?.Cancel();
+
+        using (CancellationTokenSource localCTS = new CancellationTokenSource())
+        {
+            //give future calls a chance to cancel us; it is now safe to replace
+            //any previous value of cts, since we already cancelled it above
+            cts = localCTS;
+            try
+            {
+                if (String.IsNullOrEmpty(tokenAddress)) return;
+                if (string.IsNullOrEmpty(pageNumber))
+                {
+                    pageNumber = "1";
+                }
+
+                string collectionNftKey = $"collection-{tokenAddress}-page-{pageNumber}";
+                collectionNFTSlots = await AppCache.GetOrAddAsyncNonNull(collectionNftKey,
+                    async () => await LoopringGraphQLService.GetCollectionNFTs(tokenAddress, (goToPage - 1) * pageSize, cancellationToken: localCTS.Token),
+                    DateTimeOffset.UtcNow.AddMinutes(10));
+                localCTS.Token.ThrowIfCancellationRequested();
+                Dictionary<string, NftMetadata> localNFTdata = new Dictionary<string, NftMetadata>();
+                NFTdata = localNFTdata;
+                if (collectionNFTSlots != null)
+                {
+                    StateHasChanged();
+                    foreach (var slot in collectionNFTSlots!)
+                    {
+                        string nftMetadataLinkCacheKey = $"collection-{tokenAddress}-nftMetadataLink-{slot.nftID}";
```

Comment:
What's the point of *not* reusing the caches for the meta data?

Reply:
fixed in [b23e588](https://github.com/fudgebucket27/Lexplorer/pull/182/commits/b23e5882b6733f205a1399f7bf2329f92096a773)